packages/kit/src/exports/index.js

```diff
@@ -65,9 +93,11 @@ export function text(body, init) {
 }
 
 /**
- * Generates an `ActionFailure` object.
- * @param {number} status
- * @param {Record<string, any> | undefined} [data]
+ * Create an `ActionFailure` object.
+ * @template {Record<string, any> | undefined} T
+ * @param {number} status The [HTTP status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses). Must be in the range 400-599.
+ * @param {T} [data] Data associated with the failure (e.g. validation errors)
+ * @returns {ActionFailure<T>}
```

Comment:
There's something about this change that's causing one test to fail — the `ActionResult` here...

https://github.com/sveltejs/kit/blob/f344267982002dad40f83b20f05ced5c787c6325/packages/kit/src/core/sync/write_types/test/actions/%2Bpage.server.js#L41

...has this type:

```ts
ActionResult<{
    status: number;
    data: {
        fail: string;
    } | undefined;
} | {
    success: boolean;
} | {
    id: number;
    username: string;
    profession: string;
} | {
    status: number;
    data: {
        reason: {
            error: {
                code: "VALIDATION_FAILED";
            };
        };
    } | undefined;
} | {
    ...;
}, never>
```

It should have this type:

```ts
ActionResult<{
    success: boolean;
} | {
    id: number;
    username: string;
    profession: string;
}, {
    fail: string;
} | {
    reason: {
        error: {
            code: "VALIDATION_FAILED";
        };
    };
}>
```

In other words, the `ExcludeActionFailure` and `ExtractActionFailure` types in the generated `$types.d.ts` are misfiring:

```ts
type ExcludeActionFailure<T> = T extends Kit.ActionFailure<any> ? never : T extends void ? never : T;
type ActionsSuccess<T extends Record<string, (...args: any) => any>> = { [Key in keyof T]: ExcludeActionFailure<Awaited<ReturnType<T[Key]>>>; }[keyof T];
type ExtractActionFailure<T> = T extends Kit.ActionFailure<infer X>	? X extends void ? never : X : never;
type ActionsFailure<T extends Record<string, (...args: any) => any>> = { [Key in keyof T]: Exclude<ExtractActionFailure<Awaited<ReturnType<T[Key]>>>, void>; }[keyof T];
type ActionsExport = typeof import('../../../../../../../../+page.server.js').actions
export type SubmitFunction = Kit.SubmitFunction<Expand<ActionsSuccess<ActionsExport>>, Expand<ActionsFailure<ActionsExport>>>
```

Instead of getting a `SubmitFunction<Success, Failure>` we're getting something like a `SubmitFunction<Success & Failure, never>`. I'm not sure why.